cmd/kubectl-gadget/opensnoop.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/opensnoop/types"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+var opensnoopCmd = &cobra.Command{
+	Use:   "opensnoop",
+	Short: "Trace open() system calls",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var transform func(string) string
+		switch params.OutputMode {
+		case utils.OutputModeJson: // don't print any header
```

Comment:
I was wondering if it would be useful to add also a command saying that `transform` function doesn't need to be set for `OutputModeJson` because events already arrive in JSON format so they will be printed just as they are?

Reply:
This line is not present anymore with the refactoring I did.

Reply:
nit: I would anyway add a comment to the function <gadget>TransformLine saying that it will be called to transform JSON into columns when those output modes are requested.

Reply:
Done. Btw, this code is to be refactored to avoid repeating it in all the gadgets.